src/modules/moon-pay/components/MoonPayBuyComponent.vue

```diff
@@ -612,116 +444,87 @@ export default {
       this.fetchGasPrice();
-      this.moonpayHandler
-        .getSupportedFiatToBuy(this.selectedCurrency.name)
+      this.orderHandler
+        .getSupportedFiatToBuy(this.selectedCurrency.symbol)
         .then(res => {
-          this.moonpayHandler.getFiatRatesForBuy().then(res => {
+          this.orderHandler.getFiatRatesForBuy().then(res => {
             this.currencyRates = cloneDeep(res);
             this.loading = false;
-            this.selectMoonpay();
           });
           this.fetchedData = Object.assign({}, res);
         })
         .catch(e => {
           Toast(e, {}, ERROR);
         });
+      this.getSimplexQuote();
     },
-    buy() {
-      this.moonpayHandler
-        .buy(
-          this.selectedCurrency.name,
+    getSimplexQuote() {
+      if (this.hideSimplex || !this.actualValidAddress) return;
+      this.loading = true;
+      this.simplexQuote = {};
+      this.orderHandler
+        .getSimplexQuote(
+          this.selectedCryptoName,
           this.selectedFiatName,
           this.amount,
           this.actualAddress
         )
-        .then(() => {
-          this.reset();
-          this.close();
-          this.selectedCurrency = this.defaultCurrency;
+        .then(res => {
+          this.simplexQuote = Object.assign({}, res);
+          this.loading = false;
+          this.$emit('simplexQuote', this.simplexQuote);
+          this.compareQuotes();
         })
-        .catch(err => {
-          this.reset();
-          Toast(err, {}, ERROR);
-          this.close();
-          this.selectedCurrency = this.defaultCurrency;
+        .catch(e => {
+          Toast(e, {}, ERROR);
         });
     },
-    buyFromHome() {
-      this.moonpayHandler
-        .buy(
-          this.selectedCurrency.name,
-          this.selectedFiatName,
-          this.amount,
-          this.toAddress
-        )
-        .then(() => {
-          this.reset();
-          this.close();
-          this.selectedCurrency = this.defaultCurrency;
-        })
-        .catch(err => {
-          this.reset();
-          Toast(err, {}, ERROR);
-          this.close();
-          this.selectedCurrency = this.defaultCurrency;
-        });
+    compareQuotes() {
+      const moonpayMax = this.max.moonpay;
+      // Moonpay has better rate and is not above max
+      this.showMoonpay = this.isLT(moonpayMax, this.amount) // max < amount
+        ? false
+        : this.isLT(this.simplexQuote.crypto_amount, this.moonpayCryptoAmount);
+    },
+    buy() {
+      const buyObj = {
+        cryptoToFiat: this.moonpayCryptoAmount,
+        selectedCryptoName: this.selectedCryptoName,
+        plusFeeF: this.plusFeeF,
+        includesFeeText: this.includesFeeText,
+        networkFeeText: this.networkFeeText,
+        dailyLimit: this.dailyLimit,
+        monthlyLimit: this.monthlyLimit,
+        fiatAmount: this.amount
+      };
+      this.checkMoonPayMax();
+      this.$emit('simplexQuote', this.simplexQuote);
```

Comment:
you can probably change this to 1 emit by passing an array instead `this.$emit(<event>, [...,])`